cli/src/semdep/find_lockfiles.py

```diff
@@ -110,3 +114,30 @@ def make_dependency_trie(
                 deps = list(parse_lockfile_str(lockfile.read_text(), lockfile))
                 dep_trie.insert(lockfile, deps, namespace)
     return dep_trie
+
+
+@lru_cache(maxsize=None)
+def search_path(p: Path, lockfile_pattern: str) -> Optional[Path]:
```

Comment:
Can you add function level comments on what these functions do. For this specifically probably worth clarifying that it only searches direct children of Path. 

You can prefix function name with `_` to suggest it is internal; we probably don't need this function outside of find_single_lockfile

Is there a reason to use glob over just checking `p / lockfile_pattern` exists else None? I don't think we can ever have multiples unless you allow case sensitivity.

Reply:
No there's no reason not to just if the file exists. That seems like the better thing to do.